src/libs/actions/IOU.ts

```diff
@@ -11010,6 +11024,80 @@ function putOnHold(transactionID: string, comment: string, reportID: string, sea
         },
     ];
 
+    // if the transaction thread report wasn't created before, we do it now
+    if (!initialReportID) {
+        optimisticData.push(
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.REPORT}${reportID}`,
+                value: {
+                    ...report,
```

Comment:
Why spread the report here? `Onyx.merge` should be able to update `pendingFields` without needing to modify the whole report. In the case where it's just created optimistically, we should probably use `Onyx.set` as it's more idiomatic for a plain insert.